pkg/azureutils/logger.go

```diff
@@ -0,0 +1,54 @@
+package azureutils
+
+import (
+	"fmt"
+
+	consts "sigs.k8s.io/azuredisk-csi-driver/pkg/azureconstants"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type Logger struct {
+	logr.Logger
+}
+
+func (l Logger) V(level int) Logger {
+	return Logger{l.Logger.V(level)}
+}
+
+func (l Logger) Info(msg string) {
+	l.WithCallDepth(1).Info(msg)
+}
+
+func (l Logger) Infof(msg string, params ...interface{}) {
+	l.WithCallDepth(1).Info(fmt.Sprintf(msg, params...))
+}
+
+func (l Logger) Error(err error, msg string) {
+	l.WithCallDepth(1).Error(err, msg)
+}
+
+func (l Logger) Errorf(err error, msg string, params ...interface{}) {
+	l.WithCallDepth(1).Error(err, fmt.Sprintf(msg, params...))
+}
+
+func GetRequestID(obj runtime.Object) string {
```

Comment:
This should be moved to `workflow.go` since its related to workflow and not logging.